examples/material_gallery/lib/demo/text_field_demo.dart

```diff
@@ -67,52 +63,55 @@ class TextFieldDemoState extends State<TextFieldDemo> {
       appBar: new AppBar(
         title: new Text('Text Fields')
       ),
-      body: new Block(
-        padding: const EdgeInsets.all(8.0),
-        children: <Widget>[
-          new Input(
-            hintText: 'What do people call you?',
-            labelText: 'Name',
-            errorText: _validateName(_inputs[0]),
-            value: _inputs[0],
-            onChanged: (InputValue value) { _handleInputChanged(value, 0); },
-            onSubmitted: _handleInputSubmitted
-          ),
-          new Input(
-            hintText: 'Where can we reach you?',
-            labelText: 'Phone Number',
-            errorText: _validatePhoneNumber(_inputs[1]),
-            value: _inputs[1],
-            onChanged: (InputValue value) { _handleInputChanged(value, 1); },
-            onSubmitted: _handleInputSubmitted
-          ),
-          new Row(
-            crossAxisAlignment: CrossAxisAlignment.start,
-            children: <Widget>[
-              new Flexible(
-                child: new Input(
-                  hintText: 'How do you log in?',
-                  labelText: 'New Password',
-                  hideText: true,
-                  value: _inputs[2],
-                  onChanged: (InputValue value) { _handleInputChanged(value, 2); },
-                  onSubmitted: _handleInputSubmitted
-                )
-              ),
-              new Flexible(
-                child: new Input(
-                  hintText: 'How do you log in?',
-                  labelText: 'Re-type Password',
-                  errorText: _validatePassword(_inputs[2], _inputs[3]),
-                  hideText: true,
-                  value: _inputs[3],
-                  onChanged: (InputValue value) { _handleInputChanged(value, 3); },
-                  onSubmitted: _handleInputSubmitted
-                )
+      body: new Form(
+        onSubmitted: _handleSubmitted,
+        child: new Block(
+          padding: const EdgeInsets.all(8.0),
+          children: <Widget>[
+            new Input(
+              hintText: 'What do people call you?',
+              labelText: 'Name',
+              formField: new FormField<String>(
+                // TODO(mpcomplete): replace with person#name=
+                setter:  (String val) { person.name = val; },
```

Comment:
The `person` struct isn't actually used as an input to any widgets in this example, though it certainly could be. Also, the Form class rebuilds itself and its Input widgets already, so those don't depend on a setState to rebuild.